servers/rendering/renderer_rd/storage_rd/light_storage.cpp

```diff
@@ -1535,13 +1554,12 @@ bool LightStorage::reflection_probe_instance_postprocess_step(RID p_instance) {
 	ReflectionProbeInstance *rpi = reflection_probe_instance_owner.get_or_null(p_instance);
 	ERR_FAIL_NULL_V(rpi, false);
 	ERR_FAIL_COND_V(!rpi->rendering, false);
-	ERR_FAIL_COND_V(rpi->atlas.is_null(), false);
 
 	ReflectionAtlas *atlas = reflection_atlas_owner.get_or_null(rpi->atlas);
 	if (!atlas || rpi->atlas_index == -1) {
-		//does not belong to an atlas anymore, cancel (was removed from atlas or atlas changed while rendering)
+		// Does not belong to an atlas anymore, cancel (was removed from atlas or atlas changed while rendering).
 		rpi->rendering = false;
-		return false;
+		return false; // Q: this will just keep looping, consider returning true?
```

Comment:
With the other fixes we don't get here anymore, but I still think we should change this to `return true;` to indicate there is nothing more to render. Now if there is no atlas and we somehow get this far, we would just keep calling this endlessly for this probe with nothing really happening.

Reply:
I'm not sure this will always continue rendering. I think that this will, in most cases, ensure that the probe gets updated the following frame (which it should). I have a feeling that setting to true will result in situations where the probe doesn't get updated at all if set to update once

Reply:
Indeed, if we return true here, it marks the probe as done, so if there wasn't another change that causes it to be re-queued for updating (like I've done in the situation where the atlas is rebuild) then it gets lost.

**However** if we return false, that just means it gets stuck at its current processing step and it will forever try and redo this processing step. As the failure isn't being resolved, it will never get past this point. 

The only failure scenario that I can think of that would cause this situation, is when the atlas is full. If I follow the code correctly there is a condition where a probe that was rendered some frames ago will give up its spot when a new probe comes in. But it is unclear to me whether this can happen to a probe that is still being updated over multiple frames.

So if we manage to remove a probe where we completed the first processing step (e.g. rendered the faces) but where now updating on level of the mips each frame, than that probe will forever be stuck attempting to update that mip which it can never complete because it will never attempt to redo the first step.

It really seems to me that we're dealing with logic that was written once upon a time when certain things worked differently and it would retry rendering a probe in this condition, but it is definitely broken at this point in time.

Reply:
I have removed the comment, without an MRP this is all theory and I may have done enough to prevent ever getting into this situation.
Suggest we keep this in the back of our heads in case someone runs into a broken probe :P